crates/biome_service/src/settings.rs

```diff
@@ -30,62 +30,115 @@ use indexmap::IndexSet;
 use rustc_hash::FxHashMap;
 use std::borrow::Cow;
 use std::path::{Path, PathBuf};
+use std::sync::RwLockWriteGuard;
 use std::{
     num::NonZeroU64,
     sync::{RwLock, RwLockReadGuard},
 };
 
-/// Global settings for the entire workspace
+#[derive(Debug, Default)]
+pub struct PathToSettings {
+    path: BiomePath,
+    settings: Settings,
+}
+
 #[derive(Debug, Default)]
 pub struct WorkspaceSettings {
-    /// Formatter settings applied to all files in the workspaces
-    pub formatter: FormatSettings,
-    /// Linter settings applied to all files in the workspace
-    pub linter: LinterSettings,
-    /// Language specific settings
-    pub languages: LanguageListSettings,
-    /// Filesystem settings for the workspace
-    pub files: FilesSettings,
-    /// Analyzer settings
-    pub organize_imports: OrganizeImportsSettings,
-    /// overrides
-    pub override_settings: OverrideSettings,
+    data: WorkspaceData<PathToSettings>,
+    current_workspace: WorkspaceKey,
 }
 
 impl WorkspaceSettings {
-    /// Retrieves the settings of the formatter
-    pub fn formatter(&self) -> &FormatSettings {
-        &self.formatter
+    pub fn current_settings(&self) -> &Settings {
+        debug_assert!(
+            !self.data.paths.is_empty(),
+            "You must have at least one workspace."
+        );
+        let data = self.data.paths.get(self.current_workspace).unwrap();
```

Comment:
It looks like the assertion isn't exactly guarding against failure of the `.unwrap()`, since it doesn't take the key into consideration. I think I'd write this like this:

```suggestion
        let data = self.data.paths.get(self.current_workspace)
            .expect("Workspace is not registered");
```